### PR TITLE
[BEAM-11572] Replace Assert.asserThat with MatcherAssert.assertThat across entire codebase

### PR DESCRIPTION
**Pull request by anantdamle**
JUnit [`Assert#assertThat`](https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThat(T,%20org.hamcrest.Matcher)) has is deprecated and needs to be replaced across entire codebase with [`MatcherAssert#assertThat`](https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThat(T,%20org.hamcrest.Matcher)) as per recommendation.

@iemejia  Thanks for reviewing my previous PRs. This is a massive PR, but few line changes in each file. how should I proceed?

### CODE REVIEW

File: sdks/java/io/xml/src/test/java/org/apache/beam/sdk/io/xml/XmlIOTest.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 import static org.apache.beam.sdk.transforms.display.DisplayDataMatchers.hasDisplayItem;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matcherorg.hamcrest.MatcherAssert.assertThat;
```
Comment (ibzib) [inline sdks/java/io/xml/src/test/java/org/apache/beam/sdk/io/xml/XmlIOTest.java:22]: There's a typo here.

**Comment by ibzib** [comment]
@anantdamle Do you think we could add this to our linters to prevent future usage of the deprecated class?

**Comment by anantdamle** [comment]
Yes, that should be the case.
Can you guide me on how to add that check?

**Comment by ibzib** [comment]
> Can you guide me on how to add that check?

You can add it to our [checkstyle.xml](https://github.com/apache/beam/blob/master/sdks/java/build-tools/src/main/resources/beam/checkstyle.xml) config.

**Comment by anantdamle** [comment]
Thanks let me do that.
So your recommendation is to make it part of the linter, so that slowly all code will get updated with individual PRs or in addition to this one time large one.

File: sdks/java/io/xml/src/test/java/org/apache/beam/sdk/io/xml/XmlIOTest.java (hunk 1 above)
Comment (anantdamle) [reply sdks/java/io/xml/src/test/java/org/apache/beam/sdk/io/xml/XmlIOTest.java]: Fixed.

**Comment by anantdamle** [comment]
Added linter rule. To my surprise, found one usage!

**Comment by iemejia** [comment]
Seems there is another error in the nexmark module.  I will let you and @ibzib follow with this review (that will give me some extra time to finish other stuff). Thanks.

**Comment by anantdamle** [comment]
> Seems there is another error in the nexmark module. I will let you and @ibzib follow with this review (that will give me some extra time to finish other stuff). Thanks.
> 
> ```
> 09:11:50 > Task :sdks:java:testing:nexmark:checkstyleMain
> 09:11:50 [ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_Commit@2/src/sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/queries/NexmarkQueryModel.java:102: Using deprecated Assert.assertThat. Please use MatcherAssert.assertThat. [ForbidDeprecatedAssertThat]
> ```

Thanks @iemejia fixed that.

**Comment by anantdamle** [comment]
@ibzib there is some test failure [org.apache.beam.sdk.transforms.ParDoLifecycleTest.testTeardownCalledAfterExceptionInProcessElement](https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/15391/testReport/junit/org.apache.beam.sdk.transforms/ParDoLifecycleTest/testTeardownCalledAfterExceptionInProcessElement/)
I am unable to think of a reason for this failure.

**Comment by ibzib** [comment]
> @ibzib there is some test failure [org.apache.beam.sdk.transforms.ParDoLifecycleTest.testTeardownCalledAfterExceptionInProcessElement](https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/15391/testReport/junit/org.apache.beam.sdk.transforms/ParDoLifecycleTest/testTeardownCalledAfterExceptionInProcessElement/)
> I am unable to think of a reason for this failure.

It's most likely a flake. I have seen this test fail in the past (BEAM-11541). I will try re-running.

**Comment by ibzib** [comment]
Run Java PreCommit

**Review by ibzib — APPROVED** [review]
I looked over all the changes and everything looks good. We can merge this when all the tests have passed.

**Pull request merged**